Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,759 @@
+        ready : {
+            get : function() {
+                return this._ready;
+            }
+        },
+
+        /**
+         * Gets a promise that resolves when the primitive is ready to render.
+         * @memberof GroundPolylinePrimitive.prototype
+         * @type {Promise.<GroundPolylinePrimitive>}
+         * @readonly
+         */
+        readyPromise : {
+            get : function() {
+                return this._readyPromise.promise;
+            }
+        },
+
+        /**
+         * This property is for debugging only; it is not for production use nor is it optimized.
+         * <p>
+         * If true, draws the shadow volume for each geometry in the primitive.
+         * </p>
+         *
+         * @memberof GroundPolylinePrimitive.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        debugShowShadowVolume : {
+            get : function() {
+                return this._debugShowShadowVolume;
+            }
+        }
+    });
+
+    GroundPolylinePrimitive._initialized = false;
+    GroundPolylinePrimitive._initPromise = undefined;
+
+    /**
+     * Initializes the minimum and maximum terrain heights. This only needs to be called if you are creating the
+     * GroundPolylinePrimitive synchronously.
+     *
+     * @returns {Promise} A promise that will resolve once the terrain heights have been loaded.
+     *
+     */
+    GroundPolylinePrimitive.initializeTerrainHeights = function() {
+        var initPromise = GroundPolylinePrimitive._initPromise;
+        if (defined(initPromise)) {
+            return initPromise;
+        }
+
+        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize()
+            .then(function() {
+                GroundPolylinePrimitive._initialized = true;
+            });
+
+        return GroundPolylinePrimitive._initPromise;
+    };
+
+    // For use with web workers.
+    GroundPolylinePrimitive._initializeTerrainHeightsWorker = function() {
+        var initPromise = GroundPolylinePrimitive._initPromise;
+        if (defined(initPromise)) {
+            return initPromise;
+        }
+
+        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize('../Assets/approximateTerrainHeights.json')
+            .then(function() {
+                GroundPolylinePrimitive._initialized = true;
+            });
+
+        return GroundPolylinePrimitive._initPromise;
+    };
+
+    function createShaderProgram(groundPolylinePrimitive, frameState, appearance) {
+        var context = frameState.context;
+        var primitive = groundPolylinePrimitive._primitive;
+        var attributeLocations = primitive._attributeLocations;
+
+        var vs = primitive._batchTable.getVertexShaderCallback()(PolylineShadowVolumeVS);
+        vs = Primitive._appendShowToShader(primitive, vs);
+        vs = Primitive._appendDistanceDisplayConditionToShader(primitive, vs);
```

Comment:
How well does distance work?  In this in the fragment shader now?

Reply:
It seems to work about as well as for regular polylines.

Reply:
Not sure how that is possible since the vertices are not where the fragments are rendered in most cases, right?

At least add something to the reference doc about how/why the distance condition is approximated.

Reply:
As far as I can tell the distance display condition code is based on the instance bounding sphere instead of actual vertex positions. Bounding spheres for polylines on terrain are pretty conservative though, and the fact that they're on terrain definitely complicates really precise distances. I'll add a note.